### PR TITLE
Biot's tensor

### PR DESCRIPTION
**Pull request by sytuannguyen**
This PR adds a function for computing Biot's tensor from the drained tiffness of the porous rock and the elastic moduli of the solid grain. This address an issues listed in #1508.  An approximation for Biot's coefficient of transversely isotropic rock is also given.

### CODE REVIEW

File: src/coreComponents/constitutive/solid/PorousSolid.hpp [hunk 1]
```diff
@@ -85,10 +85,26 @@ class PorousSolidUpdates : public CoupledSolidUpdates< SOLID_TYPE, BiotPorosity,
...
+    {
+      real64 const c11 = m_solidUpdate.getC11( k );
+      real64 const c13 = m_solidUpdate.getC13( k );
+      real64 const c33 = m_solidUpdate.getC33( k );
+      real64 const c66 = m_solidUpdate.getC66( k );
+
+      real64 const c12 = c11 - 2.0 * c66;
+
+      m_porosityUpdate.updateBiotCoefficient( k, c11, c12, c13, c33 );
+    }
```
Comment (CusiniM) [inline src/coreComponents/constitutive/solid/PorousSolid.hpp:107]: This functions gets called inside the poromechanics kernels in a loop over quadrature points nested inside a loop over elements. We need to find a way to implement this without using this `if statement`.

File: src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp [hunk 2]
```diff
@@ -89,13 +92,74 @@ class BiotPorosityUpdates : public PorosityBaseUpdates
...
+
   arrayView1d< real64 > m_biotCoefficient;
 
   real64 m_grainBulkModulus;
+
+  real64 m_grainShearModulus;
+
+  arrayView1d< real64 > m_b11; ///< (1,1) element of TI Biot's matrix
+  arrayView1d< real64 > m_b22; ///< (2,2) element of TI Biot's matrix
+  arrayView1d< real64 > m_b33; ///< (3,3) element of TI Biot's matrix
```
Comment (CusiniM) [inline src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp:162]: this biot's matrix is only used to compute the biotCoefficient. I don't think we need to store it.

File: src/coreComponents/constitutive/solid/PorousSolid.hpp (hunk 1 above)
Comment (sytuannguyen) [reply src/coreComponents/constitutive/solid/PorousSolid.hpp]: @CusiniM We may need to add a separated class for the PorousElasticTransverseIsotropic model. This is a simple fix but the code will be douplicated. I will try to implement this option, please take a look and give me your opinion

File: src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp (hunk 2 above)
Comment (sytuannguyen) [reply src/coreComponents/constitutive/solid/porosity/BiotPorosity.hpp]: @CusiniM For transversely isotropic or orthotropic rock, we should use the anisotropic Biot's tensor to update stress and porosity

**Pull request closed without merging**